### PR TITLE
SLF4J metrics reporter support

### PR DESCRIPTION
**Pull request by djjorjinho**
This request adds support for logging metrics to a Slf4j logger, optionally, instead of the console.

This comes in handy as a cheap way to do use Log collection and analysis software to broadcast notifications. 

Log collectors might be set up to read only log files produced by logging implementations and not STDOUT, thus the reason of this request.

Please let me know if there's anything else I can provide. Thank you.

### CODE REVIEW

**Comment by rhart** [comment]
Great, thanks! This will address #176. I will take a look over the weekend. 

If you're looking for something else metric related then I guess graphite reporter is another common one we're missing. Also, a review and updates to the documentation around metrics would be very welcome. 

Failing that, let me know what kind of thing you are interested in and we'll find something else.

**Comment by djjorjinho** [comment]
Thank you. I'm having a look at the Graphite reporter and thinking on how to tackle the various senders it supports. I'll send a request when I crack it.

File: ratpack-codahale-metrics/src/test/groovy/ratpack/codahale/metrics/MetricsSpec.groovy [hunk 1]
```diff
@@ -73,6 +73,7 @@ class MetricsSpec extends RatpackGroovyDslSpec {
         .jmx()
         .csv { it.reportDirectory(reportDirectory.root).reporterInterval(Duration.ofSeconds(1)) }
         .console { it.reporterInterval(Duration.ofSeconds(1)) }
+        .slf4j { it.reporterInterval(Duration.ofSeconds(1)).prefix("test.") }
```
Comment (rhart) [inline ratpack-codahale-metrics/src/test/groovy/ratpack/codahale/metrics/MetricsSpec.groovy]: We need test assertions to make sure this has worked.  It should output to the `output` variable I think as Gradle configures simple logger for SLF4J which outputs to system out.  We should be able to identify this output via the prefix

File: ratpack-codahale-metrics/src/test/groovy/ratpack/codahale/metrics/MetricsSpec.groovy [hunk 2]
```diff
@@ -476,6 +477,7 @@ class MetricsSpec extends RatpackGroovyDslSpec {
     bindings {
       module new CodaHaleMetricsModule(), {
         it.console { it.reporterInterval(Duration.ofSeconds(1)).includeFilter(".*ar.*").excludeFilter(".*bar.*") }
+        it.slf4j { it.reporterInterval(Duration.ofSeconds(1)).prefix("test.").includeFilter(".*ar.*").excludeFilter(".*bar.*") }
```
Comment (rhart) [inline ratpack-codahale-metrics/src/test/groovy/ratpack/codahale/metrics/MetricsSpec.groovy]: Same as the other test, we need to assert this is working as expected

**Comment by rhart** [comment]
Sorry this has taken so long.  Can you also add yourself as a contributor in the manual section.  Thanks.

**Comment by djjorjinho** [comment]
Not a problem. I've added the test assertions, but used a mock to print the output to sysout, using the logger() builder method in the process and making the test able to run outside of Gradle (IDE's and such.)

Let me know what you think.

**Comment by rhart** [comment]
Great. I will get to this tomorrow

**Comment by rhart** [comment]
Looks like there's a merge conflict in `ratpack-manual/src/content/chapters/99-about-the-project.md`.  Could you rebase and then this is good to go.

**Comment by djjorjinho** [comment]
I did the rebase. Let me know if you need anything else.

**Pull request merged**